Convex_hull_3/include/CGAL/Extreme_points_traits_adapter_3.h

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) 2018  GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+// You can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0+
+//
+//
+// Author(s)     : Maxime Gimeno
+
+#ifndef CGAL_EXTREME_POINTS_TRAITS_ADAPTER_3_H
+#define CGAL_EXTREME_POINTS_TRAITS_ADAPTER_3_H
+
+#include <CGAL/license/Convex_hull_3.h>
+
+#include <boost/call_traits.hpp>
+#include <CGAL/property_map.h>
+#include <CGAL/tags.h>
+#include <CGAL/representation_tags.h>
+#include <CGAL/Kernel_traits.h>
+
+#include <CGAL/convex_hull_3.h>
+#include <CGAL/result_of.h>
+
+
+
+namespace CGAL {
+
+namespace Convex_hull_impl{
+template <class F, class PointPropertyMap>
+struct Forward_functor
+  : public F
+{
+  PointPropertyMap vpm_;
+  
+  Forward_functor(const PointPropertyMap& vpm,
+                  const F& f) : F(f), vpm_(vpm) {}
+  
+  
+  template <class Vertex>
+  typename cpp11::result_of<F(const Vertex&, const Vertex&)>::type
+  operator() (const Vertex& p, const Vertex& q) const
+  {
+    return static_cast<const F*>(this)->operator()(get(vpm_,p),get(vpm_,q));
+  }
+
+  template <class Vertex>
+  typename cpp11::result_of<F(const Vertex&, const Vertex&, const Vertex&)>::type
+  operator() (const Vertex& p, const Vertex& q, const Vertex& r) const
+  {
+    return static_cast<const F*>(this)->operator()(
+          get(vpm_,p),
+          get(vpm_,q),
+          get(vpm_,r));
+  }
+
+  template <class Vertex>
+  typename cpp11::result_of<F(const Vertex&, const Vertex&, const Vertex&, const Vertex&)>::type
+  operator() (const Vertex& p, const Vertex& q, const Vertex& r, const Vertex& s) const
+  {
+    return static_cast<const F*>(this)->operator()(
+          get(vpm_,p), 
+          get(vpm_,q),
+          get(vpm_,r), 
+          get(vpm_,s));
+  }
+};
+}//end Convex_hull_impl
+template<class Base_traits,class PointPropertyMap>
+class Extreme_points_traits_adapter_3
```

Comment:
I would put the property map as first parameter and Base_traits default could be extracted from `internal::Convex_hull_3::Default_traits_for_Chull_3` with `boost::property_traits<PointPropertyMap>::reference` as `Point_3`. Documentation wise, you can copy-paste the doc from `convex_hull_3`: _If the kernel `R` of the points determined  ...._

Reply:
A helper function `make_extreme_points_traits_adapter` would be nice to have too.